src/animation/ValueAnimationControls.ts

```diff
@@ -198,29 +198,29 @@ export class ValueAnimationControls<P extends {} = {}, V extends {} = {}> {
     /**
      * Set motion values without animation.
      *
-     * @param target -
+     * @param definition
      * @param isActive -
      */
     private setValues(
-        { transition, transitionEnd, ...target }: TargetAndTransition,
+        definition: Variant,
         { isActive = new Set(), priority }: SetterOptions = {}
     ) {
+        let { target, transitionEnd } = this.resolveVariant(definition)
         target = this.transformValues({ ...target, ...transitionEnd } as any)
-
         return Object.keys(target).forEach(key => {
             if (isActive.has(key)) return
-
             isActive.add(key)
-            const targetValue = resolveFinalValueInKeyframes(target[key])
+            if (target) {
```

Comment:
Would this not always be `true` given we're iterating over `Object.keys(target)`?

Reply:
Typescript needs this otherwise it throws a `TS2532: Object is possibly 'undefined'`